CIPs/cip-137.md

```diff
@@ -18,190 +18,239 @@ An API for reading and writing events on Ceramic event streams.
 
-Same as `ceramic_recon`, but waits for *duration* amount of time in case any new events arrives at the node during this time.
+This content types returns the underlaying IPLD data structure of the event streams encoded as a CAR file. This data can be used to trustlessly verify the integrity of the event stream. 
 
-**Params:**
+The root object of the CAR file is *EventsResult* which is a convenience type containing an array of CIDs corresponding to the eventsids requested. The CAR file also includes all *Event* objects, including all IPLD blocks for these event, but no data from the previous events, e.g. DataEvents include signature envelope, event, and potentially detached payload, while TimeEvents include their entire merkle tree witness.
 
-- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
-- *duration* <integer> - the amount of time, in seconds, to wait for new data
+```verilog
+// https://developers.ceramic.network/protocol/streams/event-log/#data-event
+type Event InitEvent | DataEvent | TimeEvent
+
+type Events [&Event]
+```
 
-**Returns:**
 
-- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
 
 ---
 
+### POST `/ceramic/events`
 
+Adds a new event to the node.
 
-### `ceramic_exportRawEvents`
+#### Request body:
 
-Get the raw IPLD data of a set of events given an array of eventids. This includes all IPLD blocks for this particular event, but no data from the previous events, e.g. DataEvents include signature envelope, event, and potentially detached payload, while TimeEvents include their entire merkle tree witness.
+The request body must contain the complete event data.
 
-**Params:**
+**CAR:**
 
-- *[eventid]* <array<string>> - the events to fetch
+Content-Type:  `application/vnd.ipld.car`
 
-**Returns:**
+A CAR file where the root is an *Event* as described below.
 
-- *events* <string> - a base64 encoded CAR file containing the event blocks
+```verilog
+// https://developers.ceramic.network/protocol/streams/event-log/#data-event
+type Event InitEvent | DataEvent | TimeEvent
+```
 
----
+**JSON:**
 
+Content-Type:  `application/json`
 
+A possible convenience method for adding an event by only submitting a JWT.
 
-### `ceramic_putRawEvent`
+Only possible once we’ve migrated to use *Varsig* and can create events as plain signed JWTs.
 
-Add an event to a stream.
+```javascript
+{
+  event: "<jwt>"
+}
+```
 
-**Params:**
+#### Returns:
 
-- *event* <string> - a base64 encoded CAR file containing the event
 
-**Returns:**
 
-- *success* <boolean> - true if the event was added correctly
+**JSON:**
 
----
+- *eventid* <string> - the eventId of this event
 
 
 
-### `ceramic_putEvent`
+---
 
-Convenience method for adding an event by only submitting a JWT.
+### GET `/stats/active-ranges`
 
-Only possible once we’ve migrated to use *Varsig* and can create events as plain signed JWTs.
+A status endpoint that returns all ranges that this node is actively monitoring.
```

Comment:
what if two clients send two subscription requests with overlapping ranges.  Do those show up here as 2 ranges or 1?

For example, client 1 requests range from A - E, client 2 requests range from C - G.  Do we report that as one range from A-G or as 2 ranges (A-E and C-G)?